File: goblinframework-database/src/main/java/org/goblinframework/database/core/GoblinDatabaseConnection.java
package org.goblinframework.database.core;

import java.lang.annotation.*;

/**
 * Define which database should be connected. This annotation
 * should be presented on persistence/dao implementation class.
 */
@Documented
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface GoblinDatabaseConnection {

  /**
   * The name of database config which to be connected.
   */
  String name();

}

File: goblinframework-database/src/main/java/org/goblinframework/database/core/GoblinDatabaseConnection.kt
package org.goblinframework.database.core

/**
 * 标注在`persistence`和`dao`的实现类上，用于定义连接到哪个数据库的配置。
 */
@MustBeDocumented
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.RUNTIME)
annotation class GoblinDatabaseConnection(val name: String)
